exonum-java-binding/core/rust/src/handle/resource_manager/imp.rs

```diff
@@ -65,12 +88,15 @@ pub fn check_handle<T: 'static>(handle: Handle) {
         .expect("Unable to obtain read-lock")
         .get(&handle)
     {
-        Some(type_id) => {
+        Some(info) => {
             let actual_object_type = TypeId::of::<T>();
             assert_eq!(
-                *type_id, actual_object_type,
-                "Wrong type id for '{:X}' handle",
-                handle
+                info.object_type,
+                actual_object_type,
+                "Wrong type id for '{:X}' handle, expected '{}', actual '{}'",
```

Comment:
Are these included automatically in case of failure? I wonder if two _different_ types might have the same string representation (type_name)?

Reply:
Yes, they [can](https://doc.rust-lang.org/std/any/fn.type_name.html): 
> The type name should not be considered a unique identifier of a type; multiple types may share the same type name.

Do you suggest to print a different message if the string representation is the same?

Reply:
No, I think that would be redundant, I would just include the unique numeric ids.

Reply:
(if they are not already automagically included)

Reply:
They are included. The error message will look this way:
```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `TypeId { t: 16712579856250238426 }`,
 right: `TypeId { t: 7392132643490421655 }`: Wrong type id for '1' handle, expected 'u8', actual 'u16'', src/main.rs:15:5
```
Though it is a little bit messy. Would you prefer panic instead of assertion with custom formatting?

Reply:
Looks good!
> Would you prefer panic instead of assertion with custom formatting?

Not sure I understand what you mean by that ;-) Making a panic with a fully controlled error message?

Reply:
Yes, something like that:
```rust
if info.object_type != actual_object_type {
    panic!(
        "Wrong type id for '{:X}' handle, expected '{}' ({}), actual '{}' ({})",
        handle,
        info.type_name,
        info.object_type,
        any::type_name::<T>(),
        actual_object_type,
    );
}
```

Reply:
I see, thanks! As we've already got the code, I'd go with it: it is better.